br/pkg/lightning/mydump/loader.go

```diff
@@ -357,64 +357,67 @@ func (s *mdLoaderSetup) route() error {
 
 	type dbInfo struct {
 		fileMeta SourceFileMeta
-		count    int
+		count    int // means file count(db/table/view schema and table data)
```

Comment:
Is it easy to add a test for `count`?

Reply:
dbInfo is a internal struct, so it is difficult to unit test this when the `MDLoader` interface is not changed, do you have any good suggestions?

Reply:
In previous implementation, count may be negative. Can we add check for this in code?  Perhaps panic when it is negative.

Reply:
What about adding test case for `route`? For instance, u can specify the number of `db-schema-file`, `table-schema-file`, `sql-file`, and `view-schema-file`, and the rules of routing, including setting schema only, setting both schemas and tables, etc, and finally check if the `s.dbSchemas` meets your expectation. (Lots of test cases to do😭)

Reply:
add a db count check in https://github.com/pingcap/tidb/pull/33484/commits/e5a127ccf66c82f0a9ddf51ecdb541252fbf27e5

> What about adding test case for route
most case is testd in TestRouter, adding a separate test for `route` seems repetitive?